habitat/core/batched_env.py

```diff
@@ -15,6 +15,85 @@
 
 import torch  # isort:skip # noqa: F401  must import torch before importing bps_pytorch
 
+from habitat.utils.geometry_utils import  quaternion_rotate_vector 
+from habitat.tasks.utils import cartesian_to_polar
+
+from habitat_sim.utils.common import  quat_from_magnum
+
+
+class StateSensorConfig:
+    def __init__(self, config_key, shape, obs_key):
+        self.config_key = config_key
+        self.shape = shape
+        self.obs_key = obs_key
+    def get_obs(self, state):
+        pass
+
+def _get_spherical_coordinates(source_position, goal_position, source_rotation):
+    direction_vector = goal_position - source_position
+    source_rotation = quat_from_magnum(source_rotation)
+    direction_vector_agent = quaternion_rotate_vector(
+        # source_rotation.inverse(), direction_vector
+        source_rotation.inverse(), direction_vector
+    )
+    
+    _, phi = cartesian_to_polar(
+            -direction_vector_agent[2], direction_vector_agent[0]
+        )
+    theta = np.arccos(
+        direction_vector_agent[1]
+        / np.linalg.norm(direction_vector_agent)
```

Comment:
FYI Magnum has a lot of vector/matrix operations for their types and it will be good for you to eventually ramp up on them. However, if you're more comfortable with numpy linear algebra, by all means use that for now!

Reply:
I used this code: https://github.com/facebookresearch/habitat-lab/blob/main/habitat/tasks/nav/nav.py#L175 We can eventually make it with magnum if it is faster.